### DISCUSSION TITLE
Changing the maximum input length of a field to a number higher than the length of the corresponding column in the freeform_submissions table results in errors when submitting.

### DISCUSSION THREAD

**Discussion by harmputman** [solspace/craft-freeform#809] [Questions & Answers] [answered]
Changing the maximum input length of a field does not update the column length in the freeform_submissions table.

**Steps to reproduce**

1. Create a form with a text field.
2. Update the `maxLength` of the text field to 500 characters.
3. Submit the form with a string of 500 characters.
4. This results in an error "Data too long for column 'field_xx' at row..."

**Expected behavior**

When updating the `maxLength` of a field to a number greater than the length of the corresponding column in the freeform_submissions table, the length of the column in the freeform submissions table should be modified to the `maxLength` value.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 3.8.17
- Freeform Version: Pro
- Freeform Edition: 3.13.32

**Comment by kjmartens** [comment, marked as answer]
Hi @harmputman,

The `maxLength` setting is only supposed to control the `maxlength="10"` attribute (browser-level validation). If you need to allow more characters for the field in the database, you will need to update that field manually in the database and switch it from `VARCHAR` to `TEXT`.

More recent versions of Freeform 3 have improved this issue (but not retroactively), and Freeform 4+ uses `TEXT` for everything, so I assume this field was created a long time ago.

If you need further assistance adjusting this field in the database, please let me know and I'd be happy to help. 🙂
